src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/network/access/AccessProvider.kt

```diff
@@ -0,0 +1,22 @@
+package com.atlassian.performance.tools.awsinfrastructure.api.network.access
+
+import com.amazonaws.services.ec2.model.SecurityGroup
+
+/**
+ * Used to grant access to a network resource, e.g. web application.
+ * Introduced to allow restriction of anonymous access.
+ */
+interface AccessProvider {
+
+    /**
+     * @param cidr [Classless Inter-Domain Routing](https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing) format of IP and network mask
+     * @return true if it was possible to provide access, false otherwise
+     */
+    fun provideAccess(cidr: String): Boolean
```

Comment:
how about longer variable name? Without reading docs it's unclear

Reply:
I'm not sure what do you mean. Could you clarify?

Reply:
I mean the 'cidr' is not obvious without reading the doc for method.
Maybe something like 'ipWithMask' as parameter name? + example in comment not to reach wiki to check it
* @param cidr [Classless Inter-Domain Routing](https://en.wikipedia.org/wiki/Classless_Inter-Domain_Routing) format of IP and network mask
* e.g. 198.51.100.14/24

Reply:
I used the terminology that AWS API uses. Not sure if I should interpret it on our level.

Reply:
@mzyromski-atlassian what do you think about this suggestion?

Reply:
I think CIDR should be clear for a person knowing the context (AWS).